src/parser/antlr_line_buffered_input.cpp

```diff
@@ -294,10 +294,8 @@ static void bufferedInputSeek(pANTLR3_INT_STREAM is, ANTLR3_MARKER seekPoint) {
   assert(!line_buffered_input->line_buffer->isPtrBefore(
       (uint8_t*)seekPoint, input->line, input->charPositionInLine));
 
-  ssize_t count = (ssize_t)(seekPoint - (ANTLR3_MARKER)(input->nextChar));
```

Comment:
Looks plausible.

As a sanity check would you do the following for me, add a printf that count how many times the loop / is->consume() is called. Then run this on the example from the issue and respond with the numbers pre & post this commit. If the given explanation is true, you should be able to validate it.

Reply:
Unreasonably high counts for seeks were actually how I found the cause of the bug :)

Command line: `cat |  builds/bin/cvc4 --lang=smt2 | nl`

Input (entered line-by-line manually):
```
(set-option :print-success true)
(set-logic ALL)
(define-fun
 __node_init_Store_0
 ((Store.usr.Delta@0 Int))
 Bool
 (let
((X1 0))
   true))
```

Output of fixed version:
```
...
(define-fun
 __node_init_Store_0
 ((Store.usr.Delta@0 Int))
 Bool
 (let
((X1 0))
   true))    17 0
    18  0
    19  0
    20  0
    21  0
    22  0
    23  0
    24  0
    25  0
    26  0
    27  0
    28  0
    29  0
    30  0
    31  0
    32  0
    33  0
    34  4
    35  0
    36  0
    37  0
    38  0
    39  0
    40  0
    41  0
    42  0
    43  0
```

Output of master:
```
(define-fun
 __node_init_Store_0
 ((Store.usr.Delta@0 Int))
 Bool
 (let
((X1 0))
   true))    17 0
    18  0
    19  0
    20  0
    21  0
    22  0
    23  0
    24  0
    25  0
    26  0
    27  0
    28  0
    29  0
    30  0
    31  0
    32  0
    33  0
    34  974
```